doc/CMakeLists.txt

```diff
@@ -73,14 +76,18 @@ include(${ZEPHYR_BASE}/cmake/zephyr_module.cmake)
 # - ${name}-linkcheck: Run Sphinx "linkcheck" target.
 # - ${name}-clean: Clean build artifacts.
 #
-function(add_docset name env)
+function(add_docset name env dodgy)
   set(DOCSET_CFG_DIR ${CMAKE_CURRENT_LIST_DIR}/${name})
   set(DOCSET_BUILD_DIR ${CMAKE_BINARY_DIR}/${name})
   set(DOCSET_SRC_DIR ${CMAKE_BINARY_DIR}/${name}/src)
   set(DOCSET_HTML_DIR ${CMAKE_BINARY_DIR}/html/${name})
   set(DOCSET_MAKE_DIRS ${DOCSET_BUILD_DIR};${DOCSET_SRC_DIR})
   set(DOCSET_CLEAN_DIRS ${DOCSET_BUILD_DIR};${DOCSET_HTML_DIR})
 
+  if(NOT dodgy)
+    set(EXTRA_ARGS ${SPHINXOPTS_EXTRA})
```

Comment:
when having a function then it's generally good practice to pass needed settings as arguments and not relying on global / cache variable. Exception being CMake defined variables, such as CMAKE_CURRENT_LIST_DIR.

In this case use could use `cmake_parse_arguments()` and then define `SPHINXOPTS_EXTRA` as a multi value arg.

This will allow you to do:
```
add_docset(zephyr "${zephyr_env}") # No extra `SPHINXOPTS_EXTRA` should be passed.

 
add_docset(tfm "${tfm_env}" SPHINXOPTS_EXTRA ${SPHINXOPTS_EXTRA}) # We need extra options here.
```

This principle is also more future proof as it will also allow you to this, if needed in the future:
```
add_docset(new_docset "${docset_env}" SPHINXOPTS_EXTRA --special-arg) # We need --special-arg here.
```

Reply:
I have used `cmake_parse_arguments` to parse the dodgy flag, so it's not mandatory. Since this is really a temporary hack until TF-M is _fixed_, I prefer to leave it as it is.